suc/pkg/host/common.go

```diff
@@ -0,0 +1,151 @@
+package host
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultWinsPath                    = "c:\\Windows\\wins.exe"
+	defaultWinsUsrLocalBinPath         = "c:\\usr\\local\\bin\\wins.exe"
+	defaultConfigDir                   = "c:\\etc\\rancher\\wins"
+	fileOperationAttempts              = 5
+	fileOperationAttemptDelayInSeconds = 5
+
+	// skipBinaryUpgradeEnvVar prevents the suc image from attempting to upgrade the wins binary.
+	// This is primarily used in CI, to allow for test cases to run without having to completely
+	// install rancher-wins.
+	skipBinaryUpgradeEnvVar = "CATTLE_WINS_SKIP_BINARY_UPGRADE"
+)
+
+// getRancherWinsVersionFromBinary executes the wins.exe binary located at 'path' and passes the '--version'
+// flag. The release version or commit hash is returned. If the binary returns unexpected output,
+// was built with a dirty commit, or does not exist, an error will be returned.
+func getRancherWinsVersionFromBinary(path string) (string, error) {
+	if path == "" {
+		return "", fmt.Errorf("must specify a path")
+	}
+
+	_, err := os.Stat(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return "", fmt.Errorf("provided path (%s) does not exist", path)
```

Comment:
Just out of curiosity, what is the purpose for returning this separately as opposed to just using the following return?

Reply:
It's been a while since i first raised this PR, but iirc this resulted in a more useful/readable error message when inspecting the SUC pod logs